src/main/java/org/mifos/processor/bulk/zeebe/worker/SplittingWorker.java

```diff
@@ -46,7 +46,9 @@ public void setup() {
             Boolean subBatchCreated = exchange.getProperty(SUB_BATCH_CREATED, Boolean.class);
             Optional<Boolean> subBatchCreatedOptional = Optional.ofNullable(subBatchCreated);
             List<String> serverSubBatchFileList = exchange.getProperty(SERVER_SUB_BATCH_FILE_NAME_ARRAY, List.class);
-            if (subBatchCreatedOptional.isPresent() && !subBatchCreatedOptional.get() && serverSubBatchFileList.isEmpty()) {
+            Optional<List<String>> serverSubBatchFileListOptional = Optional.ofNullable(serverSubBatchFileList);
+            if (subBatchCreatedOptional.isPresent() && !subBatchCreatedOptional.get() &&
```

Comment:
looks like the optional wrapping is a bit of overhead here

Reply:
done